tests/models/bloom/test_modeling_bloom.py

```diff
@@ -377,15 +377,34 @@ def test_model_from_pretrained(self):
     @slow
     @require_torch_gpu
     def test_simple_generation(self):
+        # This test is a bit flaky. For some GPU architectures, pytorch sets by default allow_fp16_reduced_precision_reduction = True and some operations
+        # do not give the same results under this configuration, especially torch.baddmm and torch.bmm. https://pytorch.org/docs/stable/notes/numerical_accuracy.html#fp16-on-mi200
+        # We set allow_fp16_reduced_precision_reduction = True. Please see: https://pytorch.org/docs/stable/notes/cuda.html#reduced-precision-reduction-in-fp16-gemms
+        # This discrepancy is observed only when using small models and seems to be stable for larger models.
+        # Our conclusion is that these operations are flaky for small inputs but seems to be stable for larger inputs (for the functions `baddmm` and `bmm`), and therefore for larger models.
+
+        # Here is a summary of an ablation study of our observations
+        # EXPECTED_OUTPUT = "I enjoy walking with my cute dog, and I love to watch the kids play. I am a very active person, and I am a very good listener. I am a very good person, and I am a very good person. I am a"
+        # 350m + allow_fp16_reduced_precision_reduction = False  + torch.bmm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = False  + torch.baddm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = True  + torch.baddm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = True  + torch.bmm  ==> FAIL
+
+        # EXPECTED_OUTPUT = "I enjoy walking with my cute dog, but I also enjoy hiking, biking, and swimming. I love to cook and bake. I love to cook and bake. I love to cook and bake. I love to cook and bake. I love"
+        # >=760m + allow_fp16_reduced_precision_reduction = True  + torch.baddm  ==> PASS  (for use_cache=True and use_cache=False)
+        # >=760m + allow_fp16_reduced_precision_reduction = True  + torch.bmm  ==> PASS
+        # >=760m + allow_fp16_reduced_precision_reduction = False  + torch.bmm  ==> PASS
+
```

Comment:
Looks like no mentioning of `# >=760m + allow_fp16_reduced_precision_reduction = False + torch.baddm`? (But I guess it pass 😄 )

However, this seems suggesting `350m + allow_fp16_reduced_precision_reduction = True + torch.bmm` is the only failing case, and we have a comment 

https://github.com/huggingface/transformers/pull/17866#discussion_r921825354

??

Reply:
I think that you are right, @NouamaneTazi I feel that we should use `badddm` instead no?

Reply:
It was replaced by `bmm` because `350m + allow_fp16_reduced_precision_reduction = True  + torch.bmm  ==> PASS` is the only one that passed the test `test_batch_generation_padd`

Btw I just edited the comment you referenced. I linked the wrong test before. Here's the correct link: https://github.com/younesbelkada/transformers/blob/eb86c4369d465d6181a865e43fd198b28c0b7a02/tests/models/bloom/test_modeling_bloom.py#L450

Reply:
And yes @younesbelkada, we could go back to using `baddbmm` to keep the same operation used during training as mentioned by thomas

Reply:
Thanks for pointing me to the correct place 👍

Reply:
Will open a small PR and let you know @NouamaneTazi

Reply:
Created a PR: https://github.com/huggingface/transformers/pull/18175
Let's move the discussion there! @ydshieh @NouamaneTazi !